File: week_6/sentimental-readability/readability.py
text = str(input("Text : "))
letters = 0
special = 0
words = 1
sentences = 0
for i in text:
    if (i.isalpha()):
        letters += 1
    elif (i == " "):
        words += 1
    elif (i == '!' or i == '.' or i == '?'):
        sentences += 1
    else:
        special += 1

L = float(letters / words) * 100
S = float(sentences / words) * 100
index = round((0.0588 * L - 0.296 * S - 15.8))
print(0.0588 * L - 0.296 * S - 15.8)

if (index >= 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: week_2/readability/readability.c
#include<cs50.h>
#include<stdio.h>
#include<string.h>
#include<math.h>

int count_words(string text);
int count_sentences(string text);
int count_letters(string text);


int main(void)
{
    string text = get_string("Text :");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float S = (sentences / words) * 100 ;

    float L = (letters / words) * 100 ;
    float calculation = 0.0588 * L - 0.296 * S - 15.8;

    int index = round(calculation);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}
int count_letters(string text)
{
    float letters = 0 ;
    for (int i = 0 ; i < strlen(text) ; i++)
    {
        if ((text[i] >= 97 && text[i] <= 122) || (text[i] >= 65 && text[i] <= 90))
        {
            letters++;
        }

    }
    return letters;

}
int count_words(string text)
{
    int words = 0 ;
    for (int i = 0 ; i <= strlen(text) ; i++)
    {
        if (text[i] == 32 || text[i] == '\0')
        {
            words++;
        }


    }

    return words;

}

int count_sentences(string text)
{
    float sentences = 0;
    for (int i = 0 ; i < strlen(text) ; i ++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }

    }

    return sentences;

}